### PR TITLE
Clean and migrate fluid APIs of paddle.fluid.layers.control_flow

### PR DESCRIPTION
**Pull request by GhostScreaming**
### PR types
Others

### PR changes
APIs

### Describe
Clean and migrate fluid APIs of paddle.fluid.layers.control_flow and corresponding dependencies.
Remove fluid APIs:
1. BlockGuardWithCompletion
2. StaticRNNMemoryLink
3. StaticRNN
4. lod_rank_table
5. max_sequence_len
6. IfElse
7. reorder_lod_tensor_by_rank
8. is_empty

Migrate fluid APIs (from python/paddle/fluid/layers/control_flow.py to
python/paddle/static/nn/control_flow.py):
1. BlockGuard
2. WhileGuard
3. get_inputs_outputs_in_block
4. While
5. while_loop
6. case
7. switch_case

### CODE REVIEW

File: python/paddle/static/nn/control_flow.py [hunk 1]
```diff
@@ -0,0 +1,792 @@
...
+        is_test(bool, optional): A flag indicating whether execution is in test phase. Default value is False.
+        name(str, optional): The default value is None.  Normally there is no need for user to set this property.  For more information, please refer to :ref:`api_guide_Name` .
+
+    Examples 1:
+          .. code-block:: python
+
+            import paddle.fluid as fluid
+            import numpy as np
+
+            i = fluid.layers.fill_constant(shape=[1], dtype='int64', value=0)           # loop counter
```
Comment (sneaxiy) [inline python/paddle/static/nn/control_flow.py]: No fluid codes? Same above.

**Review by sneaxiy — DISMISSED** [review]

**Review by jzhang533 — COMMENTED** [review]
please cleanup `python/paddle/fluid/layers/control_flow.py`

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by jzhang533 — DISMISSED** [review]
LGTM

**Review by XieYunshen — DISMISSED** [review]
LGTM
单测删除

**Review by XieYunshen — DISMISSED** [review]

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by jzhang533 — DISMISSED** [review]

**Review by YuanRisheng — DISMISSED** [review]

**Review by zhwesky2010 — DISMISSED** [review]
LGTM for change parallel_UT

**Review by XieYunshen — APPROVED** [review]

**Review by YuanRisheng — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Pull request merged**